core/trino-main/src/main/java/io/trino/connector/system/MaterializedViewSystemTable.java

```diff
@@ -0,0 +1,139 @@
+        implements SystemTable
+{
+    private static final ConnectorTableMetadata TABLE_DEFINITION = tableMetadataBuilder(
+            new SchemaTableName("metadata", "materialized_views"))
+            .column("catalog_name", createUnboundedVarcharType())
+            .column("schema_name", createUnboundedVarcharType())
+            .column("name", createUnboundedVarcharType())
+            .column("storage_catalog", createUnboundedVarcharType())
+            .column("storage_schema", createUnboundedVarcharType())
+            .column("storage_table", createUnboundedVarcharType())
+            .column("is_fresh", BOOLEAN)
+            .column("owner", createUnboundedVarcharType())
+            .column("comment", createUnboundedVarcharType())
+            .column("definition", createUnboundedVarcharType())
+            .build();
+
+    private final Metadata metadata;
+    private final AccessControl accessControl;
+
+    @Inject
+    public MaterializedViewSystemTable(Metadata metadata, AccessControl accessControl)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return TABLE_DEFINITION;
+    }
+
+    @Override
+    public RecordCursor cursor(
+            ConnectorTransactionHandle transactionHandle,
+            ConnectorSession connectorSession,
+            TupleDomain<Integer> constraint)
+    {
+        Session session = ((FullConnectorSession) connectorSession).getSession();
+        InMemoryRecordSet.Builder displayTable = InMemoryRecordSet.builder(getTableMetadata());
+
+        Optional<String> catalogFilter = tryGetSingleVarcharValue(constraint, 0);
+        Optional<String> schemaFilter = tryGetSingleVarcharValue(constraint, 1);
+        Optional<String> tableFilter = tryGetSingleVarcharValue(constraint, 2);
+
+        listCatalogs(session, metadata, accessControl, catalogFilter).keySet().forEach(catalogName -> {
+            QualifiedTablePrefix tablePrefix = tablePrefix(catalogName, schemaFilter, tableFilter);
+
+            getMaterializedViews(session, metadata, accessControl, tablePrefix).forEach((tableName, definition) -> {
+                QualifiedObjectName name = new QualifiedObjectName(tablePrefix.getCatalogName(), tableName.getSchemaName(), tableName.getTableName());
+                MaterializedViewFreshness freshness = metadata.getMaterializedViewFreshness(session, name);
+                Object[] materializedViewRow = createMaterializedViewRow(name, freshness, definition);
+
+                displayTable.addRow(materializedViewRow);
+            });
+        });
+
+        return displayTable.build().cursor();
+    }
+
+    protected Object[] createMaterializedViewRow(
```

Comment:
should be `private static`

Reply:
I intentionally left this method as non-static, otherwise it cannot be overridden if we want to extend `MaterializedViewSystemTable` to map over richer SEP fields.

Reply:
I think we can think about extensions when they are needed. These could be in a form of new unique table.
Hence, I would skip making it extendable right now